pkg/openstack/routes.go

```diff
@@ -329,30 +332,32 @@ func (r *Routes) CreateRoute(ctx context.Context, clusterName string, nameHint s
 		defer onFailure.call(unwind)
 	}
 
-	// get the port of addr on target node.
-	port, err := getPortByIP(r.network, addr, r.networkIDs)
-	if err != nil {
-		return err
-	}
+	if r.addressAllowedPairsSupported {
```

Comment:
you can just return in case the address pairs are not supported:

```golang
if !r.allowedAddressPairs {
        klog.V(4).Infof("Route created (skipping the allowed_address_pairs update): %v", route)
        onFailure.disarm()
        return nil
}
```

and please add the same early return into the `DeleteRoute` method:

```golang
if !r.allowedAddressPairs {
        klog.V(4).Infof("Route deleted (skipping the allowed_address_pairs update): %v", route)
        onFailure.disarm()
        return nil
}
```